File: java/src/test/java/com/kcl/ParseProgramTest.java
package com.kcl;

import java.nio.file.Path;
import java.nio.file.Paths;

import org.junit.Assert;
import org.junit.Test;

import com.kcl.api.API;
import com.kcl.api.Spec.ParseProgramArgs;
import com.kcl.api.Spec.ParseProgramResult;
import com.kcl.ast.Program;
import com.kcl.util.JsonUtil;

public class ParseProgramTest {
    @Test
    public void testParseProgram() throws Exception {
        Path path = Paths.get("src", "test_data", "parse", "main.k");
        ParseProgramArgs args = ParseProgramArgs.newBuilder().addPaths(path.toString()).build();

        API apiInstance = new API();
        ParseProgramResult result = apiInstance.parseProgram(args);
        Assert.assertEquals(result.getPathsCount(), 3);
        Program program = JsonUtil.deserializeProgram(result.getAstJson());
        Assert.assertTrue(program.getRoot().contains("test_data"));
        Assert.assertEquals(program.getMainPackage().get(0).getBody().size(), 6);
    }
}

File: kotlin/src/test/kotlin/com/kcl/ParseProgramTest.kt
package com.kcl

import com.kcl.api.API
import com.kcl.api.parseProgramArgs
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.assertEquals
import java.nio.file.Paths

class ParseProgramTest {
    companion object {
        private val TEST_FILE = Paths.get("src", "test_data", "parse", "main.k").toString()
    }

    @Test
    fun testParseProgramApi() {
        val args = parseProgramArgs { paths += TEST_FILE }
        val api = API()
        val result = api.parseProgram(args)
        assertEquals(result.pathsCount, 3)
    }
}
